crates/ark/src/fixtures/dummy_frontend.rs

```diff
@@ -42,19 +42,25 @@ impl DummyArkFrontend {
         let frontend = DummyFrontend::new();
         let connection_file = frontend.get_connection_file();
 
+        // Start the kernel in this thread so that panics are propagated
+        crate::start::start_kernel(
+            connection_file,
+            vec![
+                String::from("--interactive"),
+                String::from("--vanilla"),
+                String::from("--no-save"),
+                String::from("--no-restore"),
+            ],
+            None,
+            SessionMode::Console,
+            false,
+        );
+
+        // Start the REPL in a background thread, does not return and is never joined
         stdext::spawn!("dummy_kernel", || {
-            crate::start::start_kernel(
-                connection_file,
-                vec![String::from("--no-save"), String::from("--no-restore")],
-                None,
-                SessionMode::Console,
-                false,
-            );
+            RMain::start();
         });
```

Comment:
I also wonder if we really need `R_MAIN_INIT`.

Can it _just_ be `R_INIT`/`R_STARTING`?

`RMain::is_initialized()` is only used in 2 places, both of which feel like it would be ok to wait _just a little longer_ until `R_INIT` is also set (UI comm for the "is busy" response, and connection pane)

Reply:
I'd be happy to do that simplification, I'll try it out.

Reply:
Now done. This might prevent us from establishing some state on `RMain` from other threads while R is initializing, which could take a while with some init files, but we can work this out later if needed